extensions/windows-event-log/Bookmark.cpp

```diff
@@ -50,56 +50,52 @@ Bookmark::Bookmark(const std::wstring& channel, const std::wstring& query, const
     state_manager_->set(state_map);
   }
 
-  if (!(hBookmark_ = EvtCreateBookmark(0))) {
+  if (!(hBookmark_ = unique_evt_handle{ EvtCreateBookmark(0) })) {
     LOG_LAST_ERROR(EvtCreateBookmark);
     return;
   }
 
-  const auto hEventResults = EvtQuery(0, channel.c_str(), query.c_str(), EvtQueryChannelPath);
+  const auto hEventResults = unique_evt_handle{ EvtQuery(0, channel.c_str(), query.c_str(), EvtQueryChannelPath) };
   if (!hEventResults) {
     LOG_LAST_ERROR(EvtQuery);
     return;
   }
-  const utils::ScopeGuard guard_hEventResults([hEventResults]() { EvtClose(hEventResults); });
 
-  if (!EvtSeek(hEventResults, 0, 0, 0, processOldEvents? EvtSeekRelativeToFirst : EvtSeekRelativeToLast)) {
+  if (!EvtSeek(hEventResults.get(), 0, 0, 0, processOldEvents? EvtSeekRelativeToFirst : EvtSeekRelativeToLast)) {
     LOG_LAST_ERROR(EvtSeek);
     return;
   }
 
-  DWORD dwReturned{};
-  EVT_HANDLE hEvent{};
-  if (!EvtNext(hEventResults, 1, &hEvent, INFINITE, 0, &dwReturned)) {
-    LOG_LAST_ERROR(EvtNext);
-    return;
-  }
+  const unique_evt_handle hEvent = [this,&hEventResults] {
+    DWORD dwReturned{};
+    EVT_HANDLE hEvent{ nullptr };
+    if (!EvtNext(hEventResults.get(), 1, &hEvent, INFINITE, 0, &dwReturned)) {
+      LOG_LAST_ERROR(EvtNext);
+    }
+    return unique_evt_handle{ hEvent };
+  }();
+  if (!hEvent) { return; }
```

Comment:
This feels needless to me. 
```
ok_ = hEvent && saveBookmark(hEvent.get()); 
```
Seems to do the job.

Reply:
good point, resolved in 6e93dc7